### PR TITLE
new pricing source options doc

### PR DESCRIPTION
**Pull request by jessegoodier**
New doc for easily comparing prices sources.

### CODE REVIEW

File: architecture/pricing-sources-matrix.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+| [CSV Pricing](/install-and-configure/advanced-configuration/csv-pricing.md) | Kubecost uses a user-provided CSV with granular resource prices. | Does not require cloud provider access<br>Works for on-prem clusters<br>Completely air-gapped for highly-secure environements | Mapping labels in the CSV can be tedious |
+| [AWS Spot Data Feed](/install-and-configure/install/cloud-integration/aws-cloud-integrations/aws-spot-instances.md) | Kubecost pulls spot prices from a custom feed | More accurate costs in the short-term (48 hours) | Limited value if billing-integration is configured<br>Requires additional setup with AWS |
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md:17]:
```suggestion
| [AWS Spot Data Feed](/install-and-configure/install/cloud-integration/aws-cloud-integrations/aws-spot-instances.md) | Kubecost pulls spot prices from a custom feed | More accurate costs in the short-term (48 hours) for Spot nodes | Limited value if billing-integration is configured<br>Requires additional setup with AWS |
```

File: architecture/pricing-sources-matrix.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Diagram
+
+![Cloud Provider Billing Integrated](../images/cloud-bill-diagram.png)
+
+## Detail
+
+### Cloud Provider OnDemand API
+
+[AWS EC2](https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-east-2/index.json)
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md:27]: Do we want to call out the Azure, GCP, and Alibaba APIs that are used?

File: architecture/pricing-sources-matrix.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Kubecost Pricing Sources
+
+## Overview
+
+There are multiple ways that Kubecost can be configured to pull in pricing data. This document outlines the different options and how to configure them.
+
+## Pricing sources matrix
+
+* Note: with the exception of `Cloud Provider-Billing Integrated`, all pricing sources are configured per-cluster
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md:9]: @bstuder99 Do we want to create this as a GitBook `hint` element?

File: architecture/pricing-sources-matrix.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+| [AWS Spot Data Feed](/install-and-configure/install/cloud-integration/aws-cloud-integrations/aws-spot-instances.md) | Kubecost pulls spot prices from a custom feed | More accurate costs in the short-term (48 hours) for Spot nodes | Limited value if billing-integration is configured<br>Requires additional setup with AWS |
+| [Azure Rate Card](../install-and-configure/install/cloud-integration/azure-out-of-cluster/azure-config.md) | Kubecost pulls Azure Rate Card API | More accurate Azure costs in the short-term (48 hours) | Limited value if billing-integration is configured<br>Requires additional setup with Azure |
+| [Alibaba](../install-and-configure/install/provider-installations/alibaba-install.md) | Kubecost pulls Alibaba Rate Card API | Currently the only method to retrieve Alibaba resource pricing |More accurate costs<br>Requires additional setup with Alibaba |
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md]: Suggest making multi-sentence answers either an unordered list (Can you do that in a table?) or adding proper punctuation to help delineate the details.

File: architecture/pricing-sources-matrix.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+![Cloud Provider Billing Integrated](../images/cloud-bill-diagram.png)
+
+## Details
+
+### Cloud Provider OnDemand API
+
+- [Alibaba](../install-and-configure/install/provider-installations/alibaba-install.md)
+- [AWS EC2](https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-east-2/index.json)
+- [Azure Rate Card](../install-and-configure/install/cloud-integration/azure-out-of-cluster/azure-config.md)
+
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md:34]: Add GCP pricing API link?

File: architecture/pricing-sources-matrix.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+### Custom Pricing
+
+Set via your *values-kubecost.yaml*, or in the Kubecost UI by going to *Settings*, then under 'Pricing', toggling on 'Enable Custom Pricing', then making adjustments per group:
+
+![UI Custom Pricing Screenshot](/images/custompricing.png)
+
+Prices are monthly. `storage` and miscellaneous network metrics are all per GB.
+
+```yaml
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md:43]: Should we call out that these are the helm values?
@bstuder99 Can we just name the code block as the example `values-kubecost.yaml` mentioned above?

File: architecture/pricing-sources-matrix.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+### Cloud Provider-Billing Integrated
+
+- [Cloud Provider Billing for Reconciliation and Out-of-Cluster Spend](/install-and-configure/install/cloud-integration/README.md)
+  - [AWS Cost and Usage Report](/install-and-configure/install/cloud-integration/aws-cloud-integrations/aws-cloud-integrations.md)
+  - [Azure Cost Export](/install-and-configure/install/cloud-integration/azure-out-of-cluster/azure-out-of-cluster.md)
+  - [Google BigQuery Export](/install-and-configure/install/cloud-integration/gcp-out-of-cluster/README.md)
+
+### CSV Pricing
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md:67]: Is this section necessary? It feels sparse, plus the detail is already in the other CSV pricing doc. Can we just link the table directly there?

**Review by srpomeroy — CHANGES_REQUESTED** [review]
Loving this more detail single source of truth for pricing sources!

A few minor nits and suggestions.

File: architecture/pricing-sources-matrix.md (hunk 5 above)
Comment (srpomeroy) [reply architecture/pricing-sources-matrix.md:34]:
```suggestion
- [GCP Cloud Billing API - On-Demand VMs](https://cloud.google.com/billing/docs/reference/rest/v1/services.skus/list)
```

File: architecture/pricing-sources-matrix.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+![Cloud Provider Billing Integrated](../images/cloud-bill-diagram.png)
+
+## Details
+
+### Cloud Provider OnDemand API
+
+- [Alibaba](../install-and-configure/install/provider-installations/alibaba-install.md)
+- [AWS EC2](https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-east-2/index.json)
+- [Azure Rate Card](../install-and-configure/install/cloud-integration/azure-out-of-cluster/azure-config.md)
+
```
Comment (srpomeroy) [inline architecture/pricing-sources-matrix.md:34]: We should add the Azure retail prices API as well.

```suggestion
- [Azure Retail Prices](https://learn.microsoft.com/en-us/rest/api/cost-management/retail-prices/azure-retail-prices)
```

**Review by srpomeroy — CHANGES_REQUESTED** [review]

File: architecture/pricing-sources-matrix.md (hunk 4 above)
Comment (jessegoodier) [reply architecture/pricing-sources-matrix.md]: I'll leave formattting to Brett.

Comment (jessegoodier) [reply architecture/pricing-sources-matrix.md]: @bstuder99 what are your thougts on the bullets periods at end of sentences?

File: architecture/pricing-sources-matrix.md (hunk 7 above)
Comment (jessegoodier) [reply architecture/pricing-sources-matrix.md]: killed it.

File: architecture/pricing-sources-matrix.md (hunk 8 above)
Comment (jessegoodier) [reply architecture/pricing-sources-matrix.md:37]: reworked it. Goal was to link to additional configuration. Moved reference pricing to the bottom, though I could take it out also.

**Review by srpomeroy — DISMISSED** [review]

**Review by bstuder99 — DISMISSED** [review]

**Pull request merged**